### ISSUE TITLE
BatchNorm updates never run?

### ISSUE THREAD

**Issue by AshishBora** [meetshah1995/tf-3dgan#7]
From https://www.tensorflow.org/api_docs/python/tf/contrib/layers/batch_norm :

> Note: When is_training is True the moving_mean and moving_variance need to be updated, by default the update_ops are placed in tf.GraphKeys.UPDATE_OPS so they need to be added as a dependency to the train_op, example:
> ```python
> update_ops = tf.get_collection(tf.GraphKeys.UPDATE_OPS)
> with tf.control_dependencies(update_ops):
>    train_op = optimizer.minimize(loss)
> ```

**Comment by meetps** [comment]
When using batch_norm as a layer from the contribs, the `tf.global_variables_initializer()` takes care of adding the UPDATE_OPS to the computational flow graph, you can simply set is_train= True, during test and see the difference or perhaps view the DAG in tensorboard.
